### ISSUE TITLE
delete: cannot locate 10.164.183.252 using Catalina on R6400v2

### ISSUE THREAD

**Issue by mikefreeman-uk** [jclehner/nmrpflash#36]
Hi Joseph,

1. Compiled and installed nmrpflash on Catalina. Got some errors but I figure that since it did compile, the result is still good.

`make`

`fatal: not a git repository (or any of the parent directories): .git
cc -c -Wall -g -DNMRPFLASH_VERSION=\"\" nmrp.c -o nmrp.o
fatal: not a git repository (or any of the parent directories): .git
cc -c -Wall -g -DNMRPFLASH_VERSION=\"\" tftp.c -o tftp.o
fatal: not a git repository (or any of the parent directories): .git
cc -c -Wall -g -DNMRPFLASH_VERSION=\"\" ethsock.c -o ethsock.o
fatal: not a git repository (or any of the parent directories): .git
cc -c -Wall -g -DNMRPFLASH_VERSION=\"\" main.c -o main.o
fatal: not a git repository (or any of the parent directories): .git
cc -c -Wall -g -DNMRPFLASH_VERSION=\"\" util.c -o util.o
fatal: not a git repository (or any of the parent directories): .git
cc -Wall -g -DNMRPFLASH_VERSION=\"\" -o nmrpflash nmrp.o tftp.o ethsock.o main.o util.o -lpcap`

2. Downloaded oldest firmware image, R6400v2-V1.0.2.14_1.0.7.chk from Netgear as used successfully used by user LP128 on the Netgear Community Forum: https://community.netgear.com/t5/Nighthawk-WiFi-Routers/R6400v2-bricked-after-firmware-update/m-p/1802293

3. Ran nmrpflash, with the following output...

`sudo nmrpflash -i en0 -f R6400v2-V1.0.2.14_1.0.7.chk -v`

`Adding 10.164.183.253 to interface en0.
Advertising NMRP server on en0 ... \
delete: cannot locate 10.164.183.252
Received configuration request from a0:40:a0:94:3c:a6.
Sending configuration: 10.164.183.252/24.
Timeout while waiting for TFTP_UL_REQ.
10.164.183.252 (10.164.183.252) deleted`

This response came all at once within 3 seconds, as soon as the router connected.

Any help would be much appreciated. 

Many thanks,
Mike

**Comment by jclehner** [comment]
Did you connect the router directly to your computer, or were there any devices in between? Also, what IP addresses are used on the other interfaces? The output of `ifconfig -a` would be helpful. Also, a `tcpdump` while running `nmrpflash` might contain some clues:

```
$ sudo tcpdump -i en0 -w nmrpflash.pcap
```

You can send me the `.pcap` file via email, if you don't want to post it here!

**Comment by holderl5** [comment]
Greetings,

I am hitting the exact same problem with an R6400V2.  I am using a Macbook with a thunderbolt ethernet on Catalina.

`$ sudo ./nmrpflash -vvv -i en4 -f R6400v2-V1.0.2.14_1.0.7/R6400v2-V1.0.2.14_1.0.7.chk 
Adding 10.164.183.253 to interface en4.
Advertising NMRP server on en4 ... |
Received configuration request from cc:40:d0:12:b4:00.
Sending configuration: 10.164.183.252/24.
delete: cannot locate 10.164.183.252
Timeout while waiting for TFTP_UL_REQ.
10.164.183.252 (10.164.183.252) deleted`

`$ ifconfig -a
lo0: flags=8049<UP,LOOPBACK,RUNNING,MULTICAST> mtu 16384
	options=1203<RXCSUM,TXCSUM,TXSTATUS,SW_TIMESTAMP>
	inet 127.0.0.1 netmask 0xff000000 
	inet6 ::1 prefixlen 128 
	inet6 fe80::1%lo0 prefixlen 64 scopeid 0x1 
	nd6 options=201<PERFORMNUD,DAD>
gif0: flags=8010<POINTOPOINT,MULTICAST> mtu 1280
stf0: flags=0<> mtu 1280
en0: flags=8863<UP,BROADCAST,SMART,RUNNING,SIMPLEX,MULTICAST> mtu 1500
	options=400<CHANNEL_IO>
	ether b8:e8:56:46:77:52 
	inet6 fe80::1478:1052:5280:ed73%en0 prefixlen 64 secured scopeid 0x4 
	inet6 2607:fb90:c292:c58c:183b:894f:a51e:3264 prefixlen 64 autoconf secured 
	inet6 2607:fb90:c292:c58c:8909:206e:ec5b:9fa6 prefixlen 64 autoconf temporary 
	inet 192.168.43.184 netmask 0xffffff00 broadcast 192.168.43.255
	nd6 options=201<PERFORMNUD,DAD>
	media: autoselect
	status: active
en1: flags=8963<UP,BROADCAST,SMART,RUNNING,PROMISC,SIMPLEX,MULTICAST> mtu 1500
	options=460<TSO4,TSO6,CHANNEL_IO>
	ether 82:0f:03:03:8e:c0 
	media: autoselect <full-duplex>
	status: inactive
en2: flags=8963<UP,BROADCAST,SMART,RUNNING,PROMISC,SIMPLEX,MULTICAST> mtu 1500
	options=460<TSO4,TSO6,CHANNEL_IO>
	ether 82:0f:03:03:8e:c1 
	media: autoselect <full-duplex>
	status: inactive
bridge0: flags=8863<UP,BROADCAST,SMART,RUNNING,SIMPLEX,MULTICAST> mtu 1500
	options=63<RXCSUM,TXCSUM,TSO4,TSO6>
	ether 82:0f:03:03:8e:c0 
	Configuration:
		id 0:0:0:0:0:0 priority 0 hellotime 0 fwddelay 0
		maxage 0 holdcnt 0 proto stp maxaddr 100 timeout 1200
		root id 0:0:0:0:0:0 priority 0 ifcost 0 port 0
		ipfilter disabled flags 0x0
	member: en1 flags=3<LEARNING,DISCOVER>
	        ifmaxaddr 0 port 5 priority 0 path cost 0
	member: en2 flags=3<LEARNING,DISCOVER>
	        ifmaxaddr 0 port 6 priority 0 path cost 0
	nd6 options=201<PERFORMNUD,DAD>
	media: <unknown type>
	status: inactive
p2p0: flags=8843<UP,BROADCAST,RUNNING,SIMPLEX,MULTICAST> mtu 2304
	options=400<CHANNEL_IO>
	ether 0a:e8:56:46:77:52 
	media: autoselect
	status: inactive
awdl0: flags=8943<UP,BROADCAST,RUNNING,PROMISC,SIMPLEX,MULTICAST> mtu 1484
	options=400<CHANNEL_IO>
	ether ae:2e:c0:bc:e7:4e 
	inet6 fe80::ac2e:c0ff:febc:e74e%awdl0 prefixlen 64 scopeid 0x9 
	nd6 options=201<PERFORMNUD,DAD>
	media: autoselect
	status: active
llw0: flags=8863<UP,BROADCAST,SMART,RUNNING,SIMPLEX,MULTICAST> mtu 1500
	options=400<CHANNEL_IO>
	ether ae:2e:c0:bc:e7:4e 
	inet6 fe80::ac2e:c0ff:febc:e74e%llw0 prefixlen 64 scopeid 0xa 
	nd6 options=201<PERFORMNUD,DAD>
	media: autoselect
	status: active
utun0: flags=8051<UP,POINTOPOINT,RUNNING,MULTICAST> mtu 1380
	inet6 fe80::c95:97ab:2fa0:b26e%utun0 prefixlen 64 scopeid 0xb 
	nd6 options=201<PERFORMNUD,DAD>
utun1: flags=8051<UP,POINTOPOINT,RUNNING,MULTICAST> mtu 2000
	inet6 fe80::cbbc:ca2d:c6c0:32a2%utun1 prefixlen 64 scopeid 0xc 
	nd6 options=201<PERFORMNUD,DAD>
utun2: flags=8051<UP,POINTOPOINT,RUNNING,MULTICAST> mtu 1380
	inet6 fe80::1d90:a12c:2cc:889b%utun2 prefixlen 64 scopeid 0xd 
	nd6 options=201<PERFORMNUD,DAD>
utun3: flags=8051<UP,POINTOPOINT,RUNNING,MULTICAST> mtu 1380
	inet6 fe80::2fd3:2657:ddfc:644e%utun3 prefixlen 64 scopeid 0xe 
	nd6 options=201<PERFORMNUD,DAD>
en4: flags=8863<UP,BROADCAST,SMART,RUNNING,SIMPLEX,MULTICAST> mtu 1500
	options=50b<RXCSUM,TXCSUM,VLAN_HWTAGGING,AV,CHANNEL_IO>
	ether ac:87:a3:1b:d7:d9 
	inet6 fe80::1caa:7b5b:d22c:533b%en4 prefixlen 64 secured scopeid 0x10 
	nd6 options=201<PERFORMNUD,DAD>
	media: autoselect (1000baseT <full-duplex,flow-control,energy-efficient-ethernet>)
	status: active`

`$ ./nmrpflash -L
en0              192.168.43.184   b8:e8:56:46:77:52
p2p0             0.0.0.0          0a:e8:56:46:77:52
awdl0            0.0.0.0          ae:2e:c0:bc:e7:4e
llw0             0.0.0.0          ae:2e:c0:bc:e7:4e
en4              0.0.0.0          ac:87:a3:1b:d7:d9
bridge0          0.0.0.0          82:0f:03:03:8e:c0
en1              0.0.0.0          82:0f:03:03:8e:c0
en2              0.0.0.0          82:0f:03:03:8e:c1`

I verified en4 by comparing the MAC address reported here vs in the advanced section of the network settings.

**Comment by holderl5** [comment]
I found another post where you helped someone else, and very briefly it was working.  I wasn't expecting this and hit CTRL C because I was thinking to try something else, and now I can't replicate:
`$ sudo ./nmrpflash -vvv -a 192.168.1.1 -A 192.168.1.10 -m cc:40:d0:12:b4:00 -i en4 -f R6400v2-V1.0.2.14_1.0.7/R6400v2-V1.0.2.14_1.0.7.chk 
\Adding 192.168.1.10 to interface en4.
Advertising NMRP server on en4 ... /
Received TFTP_UL_REQ while waiting for CONF_REQ!
Received upload request without filename.
Using remote filename 'R6400v2-V1.0.2.14_1.0.7.chk'.
Uploading R6400v2-V1.0.2.14_1.0.7.chk ... << WRQ(R6400v2-V1.0.2.14_1.0.7.chk, octet)
^C
$ sudo ./nmrpflash -vvv -a 192.168.1.1 -A 192.168.1.10 -m cc:40:d0:12:b4:00 -i en4 -f R6400v2-V1.0.2.14_1.0.7/R6400v2-V1.0.2.14_1.0.7.chk 
Adding 192.168.1.10 to interface en4.
Advertising NMRP server on en4 ... \
Received configuration request from cc:40:d0:12:b4:00.
Sending configuration: 192.168.1.1/24.
delete: cannot locate 192.168.1.1
Timeout while waiting for TFTP_UL_REQ.
192.168.1.1 (192.168.1.1) deleted`

**Comment by holderl5** [comment]
ok, running, it letting it fail, and then immediately running it again seemed to work.  I see a bunch of ACKS now

**Comment by xxkryptos** [comment]
I'm not a maintainer or anything, and my knowledge of this project is limited. That said, OP says they get the following error:

```sh
fatal: not a git repository (or any of the parent directories): .git cc -c -Wall -g -DNMRPFLASH_VERSION=\"\" nmrp.c -o nmrp.o 

fatal: not a git repository (or any of the parent directories): .git cc -c -Wall -g -DNMRPFLASH_VERSION=\"\" tftp.c -o tftp.o 

fatal: not a git repository (or any of the parent directories): .git cc -c -Wall -g -DNMRPFLASH_VERSION=\"\" ethsock.c -o ethsock.o fatal: not a git repository (or any of the parent directories): .git cc -c -Wall -g -DNMRPFLASH_VERSION=\"\" main.c -o main.o fatal: not a git repository (or any of the parent directories): .git cc -c -Wall -g -DNMRPFLASH_VERSION=\"\" util.c -o util.o fatal: not a git repository (or any of the parent directories): .git cc -Wall -g -DNMRPFLASH_VERSION=\"\" -o nmrpflash nmrp.o tftp.o ethsock.o main.o util.o -lpcap
```
Does this pull in external git repos? I just used this with success which is why I'm chiming in. I did a standard `git pull`. What command was run outside of a git repo? Sorry for the long post, but here's what I find:

```
user@user-device:~/Documents/nmrpflash$ grep -r git
Makefile:VERSION := $(shell if [ -d .git ] && which git 2>&1 > /dev/null; then git describe --always | tail -c +2; else echo $$STANDALONE_VERSION; fi)
README.md:Prebuilt binaries for Linux, macOS and Windows are available [here](https://github.com/jclehner/nmrpflash/releases)
Binary file .git/index matches
.git/hooks/applypatch-msg.sample:. git-sh-setup
.git/hooks/applypatch-msg.sample:commitmsg="$(git rev-parse --git-path hooks/commit-msg)"
.git/hooks/pre-push.sample:# An example hook script to verify what is about to be pushed.  Called by "git
.git/hooks/pre-push.sample:		commit=`git rev-list -n 1 --grep '^WIP' "$range"`
.git/hooks/prepare-commit-msg.sample:# Called by "git commit" with the name of the file that has the
.git/hooks/prepare-commit-msg.sample:# The second includes the output of "git diff --name-status -r"
.git/hooks/prepare-commit-msg.sample:# into the message, just before the "git status" output.  It is
.git/hooks/prepare-commit-msg.sample:#       print "\n" . `git diff --cached --name-status -r`
.git/hooks/prepare-commit-msg.sample:# SOB=$(git var GIT_COMMITTER_IDENT | sed -n 's/^\(.*>\).*$/Signed-off-by: \1/p')
.git/hooks/prepare-commit-msg.sample:# git interpret-trailers --in-place --trailer "$SOB" "$COMMIT_MSG_FILE"
.git/hooks/pre-applypatch.sample:. git-sh-setup
.git/hooks/pre-applypatch.sample:precommit="$(git rev-parse --git-path hooks/pre-commit)"
.git/hooks/pre-rebase.sample:# The "pre-rebase" hook is run just before "git rebase" starts doing
.git/hooks/pre-rebase.sample:	topic=`git symbolic-ref HEAD` ||
.git/hooks/pre-rebase.sample:git show-ref -q "$topic" || {
.git/hooks/pre-rebase.sample:not_in_master=`git rev-list --pretty=oneline ^master "$topic"`
.git/hooks/pre-rebase.sample:only_next_1=`git rev-list ^master "^$topic" ${publish} | sort`
.git/hooks/pre-rebase.sample:only_next_2=`git rev-list ^master           ${publish} | sort`
.git/hooks/pre-rebase.sample:	not_in_topic=`git rev-list "^$topic" master`
.git/hooks/pre-rebase.sample:	not_in_next=`git rev-list --pretty=oneline ^${publish} "$topic"`
.git/hooks/pre-rebase.sample:	git rev-list ^master ^topic next
.git/hooks/pre-rebase.sample:	git rev-list ^master        next
.git/hooks/pre-rebase.sample:	git rev-list master..topic
.git/hooks/pre-commit.sample:# Called by "git commit" with no arguments.  The hook should
.git/hooks/pre-commit.sample:if git rev-parse --verify HEAD >/dev/null 2>&1
.git/hooks/pre-commit.sample:	against=$(git hash-object -t tree /dev/null)
.git/hooks/pre-commit.sample:allownonascii=$(git config --bool hooks.allownonascii)
.git/hooks/pre-commit.sample:	test $(git diff --cached --name-only --diff-filter=A -z $against |
.git/hooks/pre-commit.sample:  git config hooks.allownonascii true
.git/hooks/pre-commit.sample:exec git diff-index --check --cached $against --
.git/hooks/pre-merge-commit.sample:# Called by "git merge" with no arguments.  The hook should
.git/hooks/pre-merge-commit.sample:. git-sh-setup
.git/hooks/update.sample:# Called by "git receive-pack" with arguments: refname sha1-old sha1-new
.git/hooks/update.sample:allowunannotated=$(git config --bool hooks.allowunannotated)
.git/hooks/update.sample:allowdeletebranch=$(git config --bool hooks.allowdeletebranch)
.git/hooks/update.sample:denycreatebranch=$(git config --bool hooks.denycreatebranch)
.git/hooks/update.sample:allowdeletetag=$(git config --bool hooks.allowdeletetag)
.git/hooks/update.sample:allowmodifytag=$(git config --bool hooks.allowmodifytag)
.git/hooks/update.sample:	newrev_type=$(git cat-file -t $newrev)
.git/hooks/update.sample:			echo "*** Use 'git tag [ -a | -s ]' for tags you want to propagate." >&2
.git/hooks/update.sample:		if [ "$allowmodifytag" != "true" ] && git rev-parse $refname > /dev/null 2>&1
.git/hooks/post-update.sample:exec git update-server-info
.git/hooks/fsmonitor-watchman.sample:# (https://facebook.github.io/watchman/) with git to speed up detecting
.git/hooks/fsmonitor-watchman.sample:# 'git config core.fsmonitor .git/hooks/query-watchman'
.git/hooks/fsmonitor-watchman.sample:my $git_work_tree;
.git/hooks/fsmonitor-watchman.sample:	$git_work_tree = Win32::GetCwd();
.git/hooks/fsmonitor-watchman.sample:	$git_work_tree =~ tr/\\/\//;
.git/hooks/fsmonitor-watchman.sample:	$git_work_tree = Cwd::cwd();
.git/hooks/fsmonitor-watchman.sample:		["query", "$git_work_tree", {
.git/hooks/fsmonitor-watchman.sample:		print STDERR "Adding '$git_work_tree' to watchman's watch list.\n";
.git/hooks/fsmonitor-watchman.sample:		qx/watchman watch "$git_work_tree"/;
.git/hooks/fsmonitor-watchman.sample:		die "Failed to make watchman watch '$git_work_tree'.\n" .
.git/hooks/fsmonitor-watchman.sample:		# return the fast "everything is dirty" flag to git and do the
.git/hooks/fsmonitor-watchman.sample:		# the cost in git to look up each individual file.
.git/hooks/commit-msg.sample:# Called by "git commit" with one argument, the name of the file
.git/hooks/commit-msg.sample:# SOB=$(git var GIT_AUTHOR_IDENT | sed -n 's/^\(.*>\).*$/Signed-off-by: \1/p')
.git/logs/refs/heads/master:0000000000000000000000000000000000000000 afeff8eb52bcc1f3d3e39007046eabc06b6aa251 xxkryptos <xxkryptos@protonmail.com> 1622954172 -0500	clone: from https://github.com/jclehner/nmrpflash.git
.git/logs/refs/remotes/origin/HEAD:0000000000000000000000000000000000000000 afeff8eb52bcc1f3d3e39007046eabc06b6aa251 xxkryptos <xxkryptos@protonmail.com> 1622954172 -0500	clone: from https://github.com/jclehner/nmrpflash.git
.git/logs/HEAD:0000000000000000000000000000000000000000 afeff8eb52bcc1f3d3e39007046eabc06b6aa251 xxkryptos <xxkryptos@protonmail.com> 1622954172 -0500	clone: from https://github.com/jclehner/nmrpflash.git
.git/config:	url = https://github.com/jclehner/nmrpflash.git
.git/info/exclude:# git ls-files --others --exclude-from=.git/info/exclude
main.c:			"Source code at https://github.com/jclehner/nmrpflash\n"
```

The most git-related thing I can find in the actual code is the last entry that appears to be a version string. This would appear to be an issue with your distro's (Catalina) version of the git package, or user error. By the time you execure `make`, I would have assumed you have navigated in to the repository's directory. My first thought is that you may have forgotten to navigate in to the directory after cloning. You received *fatal* messages but somehow ended up with a working binary. This makes no sense to me.

Perhaps you tried running commands outside of the directory? Eventually navigated in and actually compiled?

I can't speak much on the 2nd part of this issue. I don't know the protocol or why this might have happened.

**Comment by jclehner** [comment]
@xxkryptos this occured when building from a source tarball, as opposed to a git checkout (fixed since #35).

Closing this due to inactivity from the original reporter.
